giskard_cicd/loaders/huggingface_inf_model.py

```diff
@@ -33,41 +42,11 @@ def query(payload):
                 "Missing Hugging Face access token. Please provide it in `HF_TOKEN` environment variable"
             )
 
-        # Allow to customize the HF API endpoint
-        # See https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hfinferenceendpoint
         hf_inference_api_endpoint = os.environ.get(
             "HF_INFERENCE_ENDPOINT", default="https://api-inference.huggingface.co"
         )
-        url = f"{hf_inference_api_endpoint}/models/{model_name}"
         headers = {"Authorization": f"Bearer {hf_token}"}
-        response = requests.post(url, headers=headers, json=payload)
-        if response.status_code != 200:
-            logger.debug(f"Request to inference API returns {response.status_code}")
-        try:
-            return response.json()
-        except Exception:
-            return {"error": response.content}
-
-    # Utitlity to extract scores
-    def extract_scores(data):
-        scores = []
-
-        if isinstance(data, dict):
-            # extract 'score' value if it exists
-            score = data.get("score")
-            if score is not None:
-                scores.append(score)
-
-            # Recursively process dictionary values
-            for value in data.values():
-                scores.extend(extract_scores(value))
-
-        elif isinstance(data, list):
-            # recursively process list elements
-            for element in data:
-                scores.extend(extract_scores(element))
-
-        return scores
+        return request_inf_api(hf_inference_api_endpoint, model_name, headers, payload)
```

Comment:
During the unpicking, the model would rely on `giskard_cicd.automation.hf_api` module to load `request_inf_api` function.
But there's no problem if we decide to always install `giskard_cicd` in the worker side.